sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/serializer/HttpResponseDecodeData.java

```diff
@@ -95,42 +96,19 @@ default boolean isReturnTypeDecodable() {
     }
 
     /**
-     * Whether the network response body should be eagerly read based on its {@link #getReturnType() returnType}.
+     * How to handle the HTTP response body. The default value is {@link ResponseBodyHandling#BUFFER}.
      * <p>
-     * The following types, including subtypes, aren't eagerly read from the network:
+     * Types whose body won't be buffered by default (including sub-types):
      * <ul>
-     * <li>BinaryData</li>
-     * <li>byte[]</li>
-     * <li>ByteBuffer</li>
-     * <li>InputStream</li>
      * <li>Void</li>
      * <li>void</li>
      * </ul>
      *
-     * cracked open and their generic types are inspected for being one of the types above.
-     *
-     * @return Whether the network response body should be eagerly read.
+     * @return A {@link ResponseBodyHandling value} indicating how the body of the resulting HTTP
+     * response should be handled.
      */
-    default boolean isResponseEagerlyRead() {
-        return SwaggerMethodParser.isResponseEagerlyRead(SwaggerMethodParser.unwrapReturnType(getReturnType()));
-    }
-
-    /**
-     * Whether the network response body will be ignored based on its {@link #getReturnType() returnType}.
-     * <p>
-     * The following types, including subtypes, ignored the network response body:
-     * <ul>
-     * <li>Void</li>
-     * <li>void</li>
-     * </ul>
-     *
-     * cracked open and their generic types are inspected for being one of the types above.
-     *
-     * @return Whether the network response body will be ignored.
-     */
-    default boolean isResponseBodyIgnored() {
-        return SwaggerMethodParser.isResponseBodyIgnored(SwaggerMethodParser.unwrapReturnType(getReturnType()));
-
+    default ResponseBodyHandling getResponseBodyHandling() {
```

Comment:
We should look into removing `HttpResponseDecodeData` as it isn't really used for anything other than testing. Just replace all usages of the type with `SwaggerMethodParser`.

Reply:
I'll do that in a separate PR :)